src/components/ArcSlider.tsx

```diff
@@ -0,0 +1,254 @@
+
+		if (!this.$slider) {
+			return;
+		}
+
+		const {
+			left,
+			top,
+			width,
+			height,
+		} = this.$slider.getBoundingClientRect();
+
+		this.x = e.pageX - (left + width / 2);
+		this.y = top + height / 2 - e.pageY;
+
+		this.setState({ isDragging: true });
+	};
+
+	handleMouseMove = (e: any) => {
+		if (this.state.isDragging && this.$slider) {
+			const { left, top, width, height } = this.$slider.getBoundingClientRect();
+
+			const sliderCenter = {
+				x: left + width / 2,
+				y: top + height / 2,
+			};
+
+			const mouseLocation = !!e.touches ?
+				{
+					x: e.touches[0].pageX,
+					y: e.touches[0].pageY,
+				} :
+				{
+					x: e.pageX,
+					y: e.pageY,
+				};
+
+			const { angleRange } = this;
+
+			// Normalize the angle to make 0 degrees the topmost point of the arc
+			let angleDegrees = getAngleBetweenPoints(sliderCenter, mouseLocation) + 90;
+
+			if (angleDegrees > 180) {
+				angleDegrees = Math.max(angleDegrees - 360, 0 - angleRange);
+			} else if (angleDegrees < 0 - angleRange) {
+				angleDegrees = 0 - angleRange;
+			} else if (angleDegrees > angleRange) {
+				angleDegrees = angleRange;
+			}
+
+			const value = (angleDegrees + angleRange) / (angleRange * 2) * 100;
+
+			this.setState({
+				innerValue: value,
+			});
+
+			if (this.props.onValueChange) {
+				this.props.onValueChange(value);
+			}
+		}
+	};
+
+	render() {
+		const { angleRange } = this;
+		const { innerValue } = this.state;
+
+		// If there is no value prop, use the innerValue so that the slider can
+		// behave like an uncontrolled input
+		const value = this.props.value == null ? innerValue : this.props.value;
+		const {
+			onValueChange,
+			children,
+			...props
+		} = this.props;
+		const color = this.props.fillColor || '#FEC400';
+		const background = this.props.background || '#707274';
+		const valueAngle = (angleRange * 2 * (value / 100)) - angleRange;
+
+		return (
+			<ArcSliderContainer {...props}>
+				<StyledSVG
+					style={{
+						cursor: this.state.isDragging ? 'grabbing' : 'grab',
+					}}
+					innerRef={(elem: any) => (this.$slider = elem)}
+					viewBox="0 0 600 600"
+					onMouseDown={this.handleMouseDown}
+					onTouchStart={this.handleMouseDown}
+				>
+					<defs>
+						<linearGradient id="rendition-arc-slider-gradient" x1="0%" y1="0%" x2="100%" y2="0%">
+							<stop offset="0%" stop-color={darken(color)} />
+							<stop offset="100%" stop-color={color} />
+						</linearGradient>
+
+						<filter xmlns="http://www.w3.org/2000/svg" id="rendition-arc-slider-dropshadow" height="130%">
+							<feGaussianBlur in="SourceAlpha" stdDeviation="1.25"/>
+							<feOffset dx="0" dy="0" result="offsetblur"/>
+							<feMerge>
+								<feMergeNode/>
+								<feMergeNode in="SourceGraphic"/>
+							</feMerge>
+						</filter>
+					</defs>
+
+					<path
+						fill="none"
+						stroke={background}
+						strokeWidth="20.5"
+						strokeLinecap="round"
+						d={describeArc(300, 300, 200, 0 - angleRange, angleRange)}
```

Comment:
NIT: How about changing some `0 - angleRange` to just the unary `- angleRange`?
It was a bit confusing to me when we just meant `-135` degrees.

Reply:
done